ahmanson-lab/Booksnake
# Onboarding

Onboarding View

Details see pdf attached here:
[Booksnake - onboarding and help schematic [2022-03-17].pdf](https://github.com/ahmanson-lab/Booksnake/files/8449248/Booksnake.-.onboarding.and.help.schematic.2022-03-17.pdf)

Areas for review - Workflow and re-organization of view hierarchy

Testing Steps:
1. Users be taken through a tutorial upon downloading and starting the app. 
2. Users swipe through tutorial slides that have animated gifs playing when appropriate
3. At the end of the tutorial, users choose to go view a map in AR view **or** go to the library view
       3.a. Users choose to view map in AR. They tap the Library button to exit AR view and go 
             to the library
       3.b. Users choose to go to Library
4. Users force quit the app. When they open the app again, the tutorial page will not appear. Users are taken directly to the Library page.

*Update:

1. Optimize the animation from mp4 to gif file, play with SDWebImage kit, no video playing bar showed:

https://user-images.githubusercontent.com/20208606/162605633-1342e8c7-87d1-4f41-b9a7-1b7a301e52ff.MP4



3. Use `UserDefault` to keep track when the user has seen the onboarding, the onboardingView won't show again after forced kill / restart the app:


https://user-images.githubusercontent.com/20208606/162605640-13d30ad4-aceb-4f01-878e-48cc07f86612.mp4


2. Fix the layout for the onboarding page:

![Simulator Screen Shot - iPad Pro (12 9-inch) (5th generation) - 2022-04-09 at 23 18 07](https://user-images.githubusercontent.com/20208606/162605595-f42b3d47-41b5-4de3-92f9-442aae7f6b20.png)
![Simulator Screen Shot - iPad Pro (12 9-inch) (5th generation) - 2022-04-09 at 23 18 09](https://user-images.githubusercontent.com/20208606/162605599-04176b64-98ab-4492-bde4-afa7cdcb63f1.png)
![Simulator Screen Shot - iPad Pro (12 9-inch) (5th generation) - 2022-04-09 at 23 18 12](https://user-images.githubusercontent.com/20208606/162605601-43f5b868-5119-464f-90c6-ef48743bd986.png)
![Simulator Screen Shot - iPad Pro (12 9-inch) (5th generation) - 2022-04-09 at 23 18 14](https://user-images.githubusercontent.com/20208606/162605602-82f08558-326b-496a-91e7-e5d6578a1a60.png)
![Simulator Screen Shot - iPad Pro (12 9-inch) (5th generation) - 2022-04-09 at 23 18 16](https://user-images.githubusercontent.com/20208606/162605603-3c6b33e2-aed4-496b-9faa-e6b3e6e6ab1e.png)
